src/SignalR/clients/csharp/Http.Connections.Client/src/Internal/WebSocketsTransport.cs

```diff
@@ -35,78 +35,109 @@ internal partial class WebSocketsTransport : ITransport
 
         public WebSocketsTransport(HttpConnectionOptions httpConnectionOptions, ILoggerFactory loggerFactory, Func<Task<string?>> accessTokenProvider)
         {
-            _webSocket = new ClientWebSocket();
             _logger = (loggerFactory ?? NullLoggerFactory.Instance).CreateLogger<WebSocketsTransport>();
+            _httpConnectionOptions = httpConnectionOptions ?? new HttpConnectionOptions();
```

Comment:
logically it shouldn't be null since `HttpConnectionFactory` does a shallow copy and passes it to `HttpConnection` that passes it to `WebSocketsTransport`. but there are 7 unit tests that are creating it manually and passing options as null like this - https://github.com/dotnet/aspnetcore/blob/main/src/SignalR/server/SignalR/test/WebSocketsTransportTests.cs (all of them except the first one)

so i wasn't sure could it have a null value if there are specific tests that are passing null